driver-core/src/main/java/com/datastax/driver/core/ArrayBackedRow.java

```diff
@@ -239,6 +242,23 @@ public InetAddress getInet(String name) {
         return getInet(metadata.getFirstIdx(name));
     }
 
+    public Token getToken(int i) {
+        if (tokenFactory == null)
+            throw new DriverInternalError("Token factory not set. This should only happen at initialization time");
+
+        metadata.checkType(i, tokenFactory.getTokenType().getName());
+
+        ByteBuffer value = data.get(i);
+        if (value == null || value.remaining() == 0)
+            return null;
+
+        return tokenFactory.deserialize(value);
+    }
+
+    public Token getToken(String name) {
+        return getToken(metadata.getFirstIdx(name));
```

Comment:
Note sure I understand that last reasoning tbh, but I tend to agree with @tolbertam here: 99.999% of the time, you'll call the `getToken` getter for a column on which you had applied the `token` function in the query. And really, that's the only useful case. That is, in theory you coul call `getToken` for any column as long as the type happens to match the one of the tokens for this partitioner, but why would you ever need that?

So I do think we should automatically add the `token` part inside the getToken call. Having to always write `getToken("token(i)")` is redundant and a bit ridiculous. Note that I'm fine making it smarter so that it work whether or not you give the "token()" part yourself or not, but imo it's not worth the complexity.